rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -317,6 +376,49 @@ impl CompactOrchestrator {
         }
     }
 
+    async fn get_metadata_segment_writer(
```

Comment:
Confused - is this redundant with the inline creation above?

Reply:
yeah this is just cleanup that I missed. Thanks!